### PR TITLE
Add guest list controller, and have a working view

### PR DESCRIPTION
**Pull request by midhatdlg**
This pull request introduces significant changes to the application, including the integration of Devise for user authentication, updates to the guest list management, and modifications to the event and guest models. Below are the most important changes:

### Authentication Integration:
* Added `devise` gem to `Gemfile` to integrate Devise for user authentication.
* Created a `User` model with Devise modules for authentication, registration, password recovery, and validation.
* Configured Devise in `config/initializers/devise.rb` with default settings and custom mailer sender.

### Guest List Management:
* Created `GuestListsController` to manage guest lists, including actions for listing guests, updating attendance, and removing guests.
* Updated `GuestsController` to include authentication and event setting before actions, and modified guest list retrieval logic. [[1]](diffhunk://#diff-d614fdd9a549a0d46130f7542776a78e30a7c6286f326405fb8471bf59915c08R2-R14) [[2]](diffhunk://#diff-d614fdd9a549a0d46130f7542776a78e30a7c6286f326405fb8471bf59915c08R36-R41)

### Model Updates:
* Modified `GuestList` model to include associations with events and gift registries.

### View Enhancements:
* Added views for managing guest lists (`index.html.erb`, `destroy.html.erb`, `update_attendance.html.erb`) to display guest information and provide actions for attendance and removal. [[1]](diffhunk://#diff-27d4adda1066a940f915df1cfb278dfab423cd42371f441c6c6547dff83fde83R1-R24) [[2]](diffhunk://#diff-f39be786863ce52aa2756dea65250c3865ad27a3df8fba27998e78182d50135bR1-R2) [[3]](diffhunk://#diff-cd49c771f028b44734fae7d8b43929b6ec3e1d4dbff01d8940bd605d30bdfd09R1-R2)
* Updated `guests/index.html.erb` to display events the user is invited to and manage guest lists.

These changes collectively enhance the application's user authentication and guest list management capabilities.

### CODE REVIEW

File: app/models/guest_list.rb [hunk 1]
```diff
@@ -20,6 +20,8 @@
 #  guest_id  (guest_id => guests.id)
 #
 class GuestList < ApplicationRecord
-  has_many :guests, dependent: :destroy
-  belongs_to :event
+  has_many :guest_lists, dependent: :destroy
```
Comment (itskoms) [inline app/models/guest_list.rb:23]: remove this

File: app/models/guest_list.rb [hunk 2]
```diff
@@ -20,6 +20,8 @@
...
+  has_many :events, through: :guest_lists
```
Comment (itskoms) [inline app/models/guest_list.rb:24]: `has_one`

File: app/models/guest_list.rb [hunk 3]
```diff
@@ -20,6 +20,8 @@
...
+  has_many :gift_registries
```
Comment (itskoms) [inline app/models/guest_list.rb:25]: `has_one`

**Comment by midhatdlg** [comment]
Outdated work

**Pull request closed without merging**